cmd/minishift/cmd/start.go

```diff
@@ -187,13 +189,21 @@ func runStart(cmd *cobra.Command, args []string) {
 		atexit.Exit(1)
 	}
 
+	automountHostfolders(host.Driver)
+
 	clusterUp(&config, ip)
 	if err := openshift.AddSudoersRoleForUser("developer"); err != nil {
 		glog.Errorln("Error adding developer user to sudoers", err)
 		atexit.Exit(1)
 	}
 }
 
+func automountHostfolders(driver drivers.Driver) {
+	if hostfolder.IsAutoMount() && hostfolder.IsHostfoldersDefined(false) {
+		hostfolder.MountHostfolders(driver)
```

Comment:
tracing through your code, I see that Driver gets passed all the way down into the guts of your host folder methods, even though in many cases you just need an ssh commander. See for example https://github.com/minishift/minishift/blob/3f19376976fa015c797ac9ead42b61668ddc8e70/cmd/minishift/cmd/openshift/view.go#L78-L78

One can mock a driver for testing purposes as well, but that's harder than the ssh commander.